src/helper/DirectHelper.sol

```diff
@@ -0,0 +1,120 @@
+    function bar() external view returns (uint256);
+    function ilk() external view returns (bytes32);
+}
+
+interface TokenLike {
+    function totalSupply() external view returns (uint256);
+    function balanceOf(address) external view returns (uint256);
+    function approve(address, uint256) external returns (bool);
+    function transfer(address, uint256) external returns (bool);
+    function scaledBalanceOf(address) external view returns (uint256);
+    function decimals() external view returns (uint8);
+}
+
+interface InterestRateStrategyLike {
+    function calculateInterestRates(
+        address reserve,
+        uint256 availableLiquidity,
+        uint256 totalStableDebt,
+        uint256 totalVariableDebt,
+        uint256 averageStableBorrowRate,
+        uint256 reserveFactor
+    ) external view returns (
+        uint256,
+        uint256,
+        uint256
+    );
+}
+
+// Helper functions for keeper bots
+contract DirectHelper {
+
+    // --- Math ---
+    function _mul(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        require(y == 0 || (z = x * y) / y == x, "DssDirectDepositAaveDai/overflow");
+    }
+    uint256 constant RAY  = 10 ** 27;
+    function _rdiv(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        z = _mul(x, RAY) / y;
+    }
+
+    // Use this to determine whether exec() should be called based on your interest rate deviation threshold
+    // This assumes normal operation, culled / global shutdown should be handled externally
+    // Also assumes no liquidity issues
+    function shouldExec(
+        address _direct,
+        uint256 interestRateTolerance
+    ) external view returns (bool) {
+        // IMPORTANT: this function assumes Vat rate of this ilk will always be == 1 * RAY (no fees).
+        // That's why this module converts normalized debt (art) to Vat DAI generated with a simple RAY multiplication or division
+        // This module will have an unintended behaviour if rate is changed to some other value.
+
+        DirectLike direct = DirectLike(_direct);
+        VatLike vat = VatLike(direct.vat());
+        TokenLike dai = TokenLike(direct.dai());
+        address adai = direct.adai();
+        TokenLike stableDebt = TokenLike(direct.stableDebt());
+        TokenLike variableDebt = TokenLike(direct.variableDebt());
+        bytes32 ilk = direct.ilk();
+        InterestRateStrategyLike interestStrategy = InterestRateStrategyLike(direct.interestStrategy());
+
+        (uint256 daiDebt,) = vat.urns(ilk, address(direct));
+        uint256 _bar = direct.bar();
+        if (_bar == 0) {
+            return daiDebt > 0;     // Always attempt to close out if we have debt remaining
+        }
+
+        (,, uint256 currVarBorrow) = interestStrategy.calculateInterestRates(
+            adai,
+            dai.balanceOf(adai),
+            stableDebt.totalSupply(),
+            variableDebt.totalSupply(),
+            0,
+            0
+        );
```

Comment:
in D3M for `LendingPoolLike` the comments indicate that we could get the current variable rate using `direct.pool.getReserveData(address(dai))`

I think this makes things similar and seems to work: https://github.com/makerdao/dss-direct-deposit/pull/23

Let me know if there is some Aave implementation detail I'm missing (I didn't really find this section in their docs, so kind of piecing things together).

Gas diff: 1472009 -> 1274197